File: our-knowledge-front/src/backend/userService.js
import {config, appFetch} from './appFetch';

export const size = 4;

export const login = (userName, email, role, onSuccess, onErrors) =>
    appFetch(`/user/login`,
        config('POST', {userName, email, role}), onSuccess, onErrors);

export const findUsers = (page, keywords, filterId, onSuccess) =>
    appFetch(`/user/list`,
        config('POST', {size, page, keywords, filterId}), onSuccess);

export const getProfile = (userId, onSuccess, onErrors) =>
    appFetch(`/user/profile`,
        config('POST', {userId}), onSuccess, onErrors);

export const updateUser = (startDate, onSuccess, onErrors) =>
    appFetch(`/user/update`,
        config('POST', {startDate}), onSuccess, onErrors);

File: OurKnowledgeBackend/src/main/java/com/example/ourknowledgebackend/service/UserService.java
package com.example.ourknowledgebackend.service;

import com.example.ourknowledgebackend.exceptions.InstanceNotFoundException;
import com.example.ourknowledgebackend.model.UserProfile;
import com.example.ourknowledgebackend.model.UserResult;
import com.example.ourknowledgebackend.model.entities.User;

import java.text.ParseException;

public interface UserService {

    User login(String userName, String email, String role);

    Block<UserResult> listUsers(int page, int size, String keywords, Long filterId) throws InstanceNotFoundException;

    UserProfile showProfile(Long profileId, Long userId) throws InstanceNotFoundException;

    void updateUser(Long userId, String startDate) throws InstanceNotFoundException, ParseException;
}
